File: app/src/java/java/com/bear2b/sampleapp/ui/view/activities/AdvancedSampleActivity.java
package com.bear2b.sampleapp.ui.view.activities;

import android.app.Fragment;
import android.app.FragmentManager;
import android.graphics.Color;
import android.os.Bundle;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import com.bear.common.sdk.BearCallback;
import com.bear.common.sdk.BearSdk;
import com.bear.common.sdk.IBearHandler;
import com.bear.common.sdk.ui.activities.main.ArActivity;
import com.bear2b.sampleapp.R;
import com.bear2b.sampleapp.ui.view.fragments.HistoryFragment;
import com.bear2b.sampleapp.ui.view.fragments.MainFragment;
import com.bear2b.sampleapp.ui.view.fragments.SamplePausingFragment;
import com.bear2b.sampleapp.ui.view.fragments.SuccessScanFragment;
import com.bear2b.sampleapp.ui.view.fragments.WebViewFragment;

import org.jetbrains.annotations.NotNull;

import java.util.List;

import androidx.annotation.IdRes;
import androidx.annotation.NonNull;

public class AdvancedSampleActivity extends ArActivity {

    private FragmentManager fragmentManager = getFragmentManager();
    public IBearHandler bearHandler = getBearHandler();

    @IdRes
    private int container = R.id.container;

    @Override
    public int getScanTimeout() {
        return 10;
    }

    @Override
    public int getScanLineColor() {
        return Color.rgb(80, 44, 108);
    }

    private BearCallback newCallback = new BearCallback() {
        @Override
        public void onMarkerWithUnsupportedAsset() {
            showAlert("onMarkerWithUnsupportedAsset");
        }

        @Override
        public void onArViewInitialized() {
            showAlert("ArView initialization completed");
            fragmentManager
                    .beginTransaction()
                    .add(container, MainFragment.newInstance())
                    .commit();
        }

        @Override
        public void onMarkerRecognized(final int i, @NonNull final List<Integer> list) {
            showAlert("onMarkerRecognized, marker id = " + i + " assets id = " + list.toString());
            replaceFragment(SuccessScanFragment.newInstance());
        }

        @Override
        public void onMarkerNotRecognized() {
            showAlert("onMarkerNotRecognized");
        }

        @Override
        public void onAssetClicked(final int i) {
            showAlert("onAssetClicked " + i);
        }

        @Override
        public void onError(@NonNull final Throwable t) {
            showAlert("Error occurred: " + t.toString());
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        bearHandler.registerBearCallback(newCallback);
        System.out.println(BearSdk.getInstance(this).getDeviceId());
    }

    @Override
    protected void onDestroy() {
        bearHandler.unregisterBearCallback(newCallback);
        super.onDestroy();
    }

    @Override
    public void inflateContentView() {
        View view = getLayoutInflater().inflate(R.layout.activity_advanced_sample, null, false);
        addContentView(view, new ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT));
    }

    @Override
    public void openWebView(@NotNull String url) {
        replaceFragment(WebViewFragment.newInstance(url));
    }

    @Override
    public void onBackPressed() {
        if (fragmentManager.getBackStackEntryCount() == 1) bearHandler.cleanArView();
        super.onBackPressed();
    }

    public void showHistoryScreen() {
        replaceFragment(HistoryFragment.newInstance());
    }

    public void showSamplePausingFragment() {
        replaceFragment(SamplePausingFragment.newInstance());
    }

    private void replaceFragment(Fragment fragment) {
        fragmentManager
                .beginTransaction()
                .replace(container, fragment)
                .addToBackStack(null)
                .commit();
    }

    @Override
    public void showAlert(@NotNull String message) {
        Toast.makeText(this, message, Toast.LENGTH_LONG).show();
    }

    @Override
    public void showAlert(int messageId) {
        showAlert(getString(messageId));
    }
}

File: app/src/kotlin/java/com/bear2b/sampleapp/ui/view/activities/AdvancedSampleActivity.kt
package com.bear2b.sampleapp.ui.view.activities

import android.graphics.Color
import android.os.Bundle
import android.view.ViewGroup.LayoutParams
import android.widget.Toast
import androidx.annotation.IdRes
import androidx.annotation.NonNull
import androidx.fragment.app.Fragment
import com.bear.common.sdk.BearCallback
import com.bear.common.sdk.BearSdk
import com.bear.common.sdk.ui.activities.main.ArActivity
import com.bear2b.sampleapp.R
import com.bear2b.sampleapp.ui.view.fragments.*
import org.jetbrains.annotations.NotNull

class AdvancedSampleActivity : ArActivity() {

    @IdRes
    private val container = R.id.container

    override val scanTimeout = 10

    override val scanLineColor = Color.rgb(80, 44, 108)

    private val newCallback = object : BearCallback {
        override fun onArViewInitialized() {
            showAlert("ArView initialization completed")
            supportFragmentManager
                    .beginTransaction()
                    .add(container, MainFragment.newInstance())
                    .commit()
        }

        override fun onMarkerRecognized(markerId: Int, @NonNull assetsId: List<Int>) {
            showAlert("onMarkerRecognized marker id = $markerId assets id = $assetsId")
            replaceFragment(SuccessScanFragment.newInstance())
        }

        override fun onMarkerNotRecognized() = showAlert("onMarkerNotRecognized")

        override fun onAssetClicked(assetId: Int) = showAlert("onAssetClicked $assetId")

        override fun onError(@NonNull error: Throwable) = showAlert("Error occurred: $error")
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        bearHandler.registerBearCallback(newCallback)
        println(BearSdk.getInstance(this).deviceId)
    }

    override fun onDestroy() {
        bearHandler.unregisterBearCallback(newCallback)
        super.onDestroy()
    }

    override fun inflateContentView() {
        val view = layoutInflater.inflate(R.layout.activity_advanced_sample, null, false)
        addContentView(view, LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT))
    }

    override fun openWebView(@NotNull url: String) = replaceFragment(WebViewFragment.newInstance(url))

    override fun onBackPressed() {
        if (supportFragmentManager.backStackEntryCount == 1) bearHandler.cleanArView()
        super.onBackPressed()
    }

    fun showHistoryScreen() = replaceFragment(HistoryFragment.newInstance())

    fun showSamplePausingFragment() = replaceFragment(SamplePausingFragment.newInstance())

    private fun replaceFragment(fragment: Fragment) {
        supportFragmentManager
                .beginTransaction()
                .replace(container, fragment)
                .addToBackStack(null)
                .commit()
    }

    override fun showAlert(message: String) {
        Toast.makeText(this, message, Toast.LENGTH_LONG).show()
    }

    override fun showAlert(messageId: Int) {
        showAlert(getString(messageId))
    }
}